srml/balances/src/lib.rs

```diff
@@ -323,55 +349,81 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
 	}
 
 	/// Register a new account (with existential balance).
+	///
+	/// This just calls appropriate hooks. It doesn't (necessarily) make any state changes.
 	fn new_account(who: &T::AccountId, balance: T::Balance) {
 		T::OnNewAccount::on_new_account(&who);
 		Self::deposit_event(RawEvent::NewAccount(who.clone(), balance.clone()));
 	}
 
+	/// Unregister an account.
+	///
+	/// This just removes the nonce and leaves an event.
 	fn reap_account(who: &T::AccountId) {
 		<system::AccountNonce<T>>::remove(who);
```

Comment:
There is `T::OnNewAccount`, this here also be `T::OnReapAccount` as well?

Reply:
You mostly care about the Free part of an account being destroyed (`T::OnFreeBalanceZero`). I don't mind adding a hook if you can come up with a use case.